services/compute/virtualmachine/virtualmachine.go

```diff
@@ -882,3 +904,21 @@ func (c *client) getVirtualMachineProxyConfiguration(proxyConfiguration *wssdclo
 		TrustedCa:  &proxyConfiguration.TrustedCa,
 	}
 }
+
+func (c *client) getZoneConfiguration(zoneConfiguration *wssdcommon.ZoneConfiguration) *compute.ZoneConfiguration {
+	if zoneConfiguration == nil || len(zoneConfiguration.Zones) == 0 {
+		return nil
+	}
+
+	zones := []compute.Zone{}
+	strinctPlacement := zoneConfiguration.StrictPlacement
+	for _, zone := range zoneConfiguration.Zones {
```

Comment:
Should we add a nil check?

Reply:
zoneConfiguration.Zones is an array of pointers i.e. 'Zones   []*ZoneReference'. So nil check is not required.